nemo/collections/tts/data/dataset.py

```diff
@@ -485,11 +486,20 @@ def add_speaker_id(self, **kwargs):
         pass
 
     def add_reference_audio(self, **kwargs):
-        assert SpeakerID in self.sup_data_types, "Please add speaker_id in sup_data_types."
-        """Add a mapping for each speaker to their manifest indexes"""
-        self.speaker_to_index_map = defaultdict(set)
-        for i, d in enumerate(self.data):
-            self.speaker_to_index_map[d['speaker_id']].add(i)
+        reference_audio_type = kwargs.pop("reference_audio_type", "same-speaker")
+        if reference_audio_type == "same-speaker":
+            assert SpeakerID in self.sup_data_types, "Please add speaker_id in sup_data_types."
+            # Add a mapping for each speaker to their manifest indexes
+            speaker_to_index_map = defaultdict(set)
+            for i, d in enumerate(self.data):
+                speaker_to_index_map[d["speaker_id"]].add(i)
+            # Random sample a reference audio from the same speaker
+            self.get_reference_for_sample = lambda sample: self.data[speaker_to_index_map[sample["speaker_id"]]]
```

Comment:
We may need random sample an index here:
```
lambda sample: self.data[random.sample(speaker_to_index_map[sample["speaker_id"]], 1)[0]]
```

Reply:
Damn, my bad. Fixed now!